internal/format/format.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package format
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-cli/internal/config"
+	"github.com/edgexfoundry/edgex-cli/internal/service"
+)
+
+type FormatType int
+
+const (
+	Debug FormatType = iota
+	JSON
+	Value
```

Comment:
What is the difference and why do we need both?

Reply:
`version` and `ping` just need to print a single value. `metrics` returns a list of strings and `config` a hierarchy. This is really there to make the output of `version` and `ping` a bit more user-friendly and just display that single value, rather than the whole JSON output. See `edgex-cli version` vs `edgex-cli metrics`